### PR TITLE
MGMT-13178: Adding order parameter and changing the count of events by severity to apply only after the filtering

### PR DESCRIPTION
**Pull request by danmanor**
As part of events pagination epic, this PR is about adding `order` parameter so the client can request the events in the desired order, and changing the counting of events by severity to apply after the filtering

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/events/event.go [hunk 1]
```diff
@@ -455,7 +471,7 @@ func (e Events) V2GetEvents(ctx context.Context, params *common.V2GetEventsParam
 		selectedCategories = append(selectedCategories, DefaultEventCategories...)
 	}
 
-	events, eventSeverityCount, err := e.queryEvents(ctx, selectedCategories, params.ClusterID, params.HostIds, params.InfraEnvID, params.Limit, params.Offset, params.Severities, params.Message, params.DeletedHosts, params.ClusterLevel)
+	events, eventSeverityCount, err := e.queryEvents(ctx, selectedCategories, params.ClusterID, params.HostIds, params.InfraEnvID, params.Limit, params.Offset, params.Order, params.Severities, params.Message, params.DeletedHosts, params.ClusterLevel)
```
Comment (rccrdpccl) [inline internal/events/event.go:474]: Not really relevant to this PR but I guess we should pass all params through instead having a long list

File: internal/events/event.go [hunk 2]
```diff
@@ -408,37 +409,51 @@ func preparePaginationParams(limit, offset *int64) (*int64, *int64) {
 	return limit, offset
 }
 
-func (e Events) queryEvents(ctx context.Context, selectedCategories []string, clusterID *strfmt.UUID, hostIds []strfmt.UUID, infraEnvID *strfmt.UUID, limit, offset *int64, severity []string, message *string, deletedHosts, clusterLevel *bool) ([]*common.Event, *common.EventSeverityCount, error) {
+func prepareOrderParam(order *string) (*string, error) {
+	if order == nil || *order == "ascending" {
+		return swag.String("asc"), nil
+	} else if *order == "descending" {
```
[the remark this answers is not in the document]
Comment (danmanor) [reply internal/events/event.go:415]: Missed that thanks

**Pull request merged**